doc/developer-guide.md

```diff
@@ -0,0 +1,260 @@
+# ddf-validation Developer's guide 
+
+## Registry
+
+First of all new rule should be added to the [rules registry](../lib/ddf-rules/registry.js).
+
+A new rule should be added next way:
+
+```javascript
+exports.MY_NEW_RULE = Symbol.for('MY_NEW_RULE');
+```
+
+Note: `Symbol.for` syntax is mandatory.
+
+Also, this rule should be added to descriptions:
+
+```javascript
+exports.descriptions = {
+  /// ...
+  [exports.MY_NEW_RULE]: 'Description for this rule',
+  /// ...
+};
+```
+
+## Rules applying explanation
+
+There are next types of rules:
+
+* General rules
+* Index rules
+* Concept rules
+* Entity rules
+* Data points rules
+
+### General rules
+
+This kind of rules is about situation when program doesn't know about Concepts, Entities and Data points.
+In fact, this kind of rule is not DDF based. For example, rule `Folder is not DDF`.
+
```

Comment:
General rules are applied to whole DDF data set. And the don't validate Concepts, Entities and Datapoints on their own.